tests/pipeline/test_manifest.py

```diff
@@ -123,15 +123,19 @@ def test_manifest_command(self, snapshot: Snapshot):
         assert result.exit_code == 0, result.stdout
         snapshot.assert_match(result.stdout, MANIFEST_YAML)
 
-    def test_python_api(self, snapshot: Snapshot):
+    def test_python_api(self):
         generator = kpops.manifest_deploy(
             RESOURCE_PATH / "manifest-pipeline" / PIPELINE_YAML,
             environment="development",
         )
         assert isinstance(generator, Iterator)
         resources = list(generator)
         assert len(resources) == 2
-        snapshot.assert_match(yaml.dump_all(resources), "resources")
```

Comment:
the new isinstance check is great, but I think we would still need some sort of snapshot testing as well. how about we just save the output of `print_yaml(manifest.model_dump)`?

Reply:
Added it :)

Reply:
I agree with you. we should probably split that up into two separate tests. In the API test we run assertions on the returned `KubernetesManifest` models and in the CLI test we do the snapshot test of the output. Let's do it in a followup then